### DISCUSSION TITLE
Secret File for Config Settings and/or Environment Variables?

### DISCUSSION THREAD

**Discussion by mrbiggred** [community/community#26307] [Actions]
<p>Hello,</p>
<p>I have an application that reads a bunch of ENV variables from a file.  Mostly API keys used during integration testing that I don’t want in the repo.  I would prefer to not have to set each of these ENV variables as a seperate secret variable.  Is there a way to save a file, say a .env file, as a secret then load that file during the run?  In my case my application will read and set the ENV variables from the file.</p>
<p>Something similar to how Jenkins lets you store secret files?</p>
<p><a href="https://jenkins.io/doc/book/using/using-credentials/" rel="nofollow noopener">https://jenkins.io/doc/book/using/using-credentials/</a></p>
<p>Thanks.</p>

**Comment by AnthonyWC** [comment]
<p>You just need to load your variables into the github virtual environment.  e.g. export in whatever languages/tools of your choices.  e.g. dotenv for javascript or in bash just _export $(&lt;.env) _inside one of your steps. (as long as they are in the same job they will be in the same environment).</p>

**Comment by mrbiggred** [comment]
<p>Hello <a class="mention" href="/anthonywc">@anthonywc</a>.   How do I get the file with my environment variables into the virtual environment if it’s not in the repository?  The file has API keys that I don’t want to add to the repository.  Thank you.</p>

**Comment by mscoutermarsh** [comment, marked as answer]
<p>Hi <a class="mention" href="/mrbiggred">@mrbiggred</a>,</p>
<p>We don’t have that capability currently.</p>
<p>We do have an API coming soon for setting secrets. That could be used for setting large numbers of secrets programmatically.</p>
<p>Sorry we don’t have an exact solution for you. Hopefully the API makes this workflow possible for you though. </p>

**Comment by mrbiggred** [comment]
<p>Thank you <a class="mention" href="/smscodeverification">@smscodeverification</a> for answing my question even it it was not he answer I wanted to hear.  Also thnk you letting me know about the upcoming API.  All the best.</p>

**Comment by wiffel** [comment]
<p>You can store your secrets inside repository encrypted with ‘gpg’ tool and decrypt them as a step in workflow job</p>
<p>a similar solution is described here <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#limits-for-secrets" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#limits-for-secrets</a></p>

**Comment by ErlanBazarov** [comment]
<p>Any updates on this issue?</p>

**Comment by jashwant** [comment]
<p>Can’t we store content of .env file in a secret variable ENV_FILE and then create a step to create .env file from secrets,</p>
<pre><code class="lang-auto">- name: Create .env file
   run:  echo "${{ secrets.ENV_FILE }}" &gt; .env
</code></pre>
<p>Then may be delete .env file later</p>

**Reply by LucBerge** [reply]
This is a correct solution.

- In you app: load your config from .env file using a lib like dotenv.
- In github action: Create a .env file and populate it

```
- name: Create .env file
   run:  |
       echo "${{ secrets.ENV_FILE }}" > .env
       echo "${{ secrets.VAR1 }}" >> .env
       echo "${{ secrets.VAR2 }}" >> .env
```

**Comment by vntech-engineering39** [comment]
<p>Any Update on this issue?</p>
